tools/lambda-promtail/lambda-promtail/s3.go

```diff
@@ -123,6 +127,37 @@ func parseS3Log(ctx context.Context, b *batch, labels map[string]string, obj io.
 	return nil
 }
 
+func parseLogLineTimestamp(log_line string) time.Time {
+	for _, regex := range timestampRegexList {
+		match := regex.FindStringSubmatch(log_line)
+		if len(match) > 0 {
+			var timestamp time.Time
+			var err error
+			var timeToParseNumber int64
+
+			//Try RFC3339 format
+			timestamp, err = time.Parse(time.RFC3339, match[1])
+			if err == nil {
+				return timestamp
+			}
+
+			//Try milliseconds/seconds format
+			timeToParseNumber, err = strconv.ParseInt(match[1], 10, 64)
+			if err == nil {
+				//https://stackoverflow.com/questions/23929145/how-to-test-if-a-given-time-stamp-is-in-seconds-or-milliseconds
+				dateNowSecs := time.Now().Unix()
+				if timeToParseNumber > dateNowSecs {
+					return time.UnixMilli(timeToParseNumber)
+				}
+				return time.Unix(timeToParseNumber, 0)
+			}
```

Comment:
we could format this as
```
if err != nil {
    continue
}
// seconds vs ms check here
```

is this "seconds or milliseconds" timestamp format specific to WAF logs?

Reply:
Ok, will do, thanks.

Milliseconds is the format for the timestamp field for AWS WAF logs.
Seconds is the format for the begin/end fields for AWS VPC Flow logs